### PR TITLE
Add documentation for PEHeader

### PR DESCRIPTION
**Pull request by MSDN-WhiteKnight**
Add documentation and example for System.Reflection.PortableExecutable.PEHeader

### CODE REVIEW

**Review by steveharter — APPROVED** [review]
Thanks!

File: xml/System.Reflection.PortableExecutable/PEHeader.xml [hunk 1]
```diff
@@ -334,14 +346,14 @@ Represents `IMAGE_DIRECTORY_ENTRY_COPYRIGHT` or `IMAGE_DIRECTORY_ENTRY_ARCHITECT
...
-        <value>To be added.</value>
+        <summary>Gets the CLI header table entry.</summary>
+        <value>The CLI header table entry.</value>
         <remarks>
           <format type="text/markdown"><![CDATA[
 
 ## Remarks
 
-Represents `IMAGE_DIRECTORY_ENTRY_COM_DESCRIPTOR`.
+The CLI header contains information and data directories used by .NET runtime. The CLI header format is defined by the [ECMA-335 specification](https://www.ecma-international.org/publications-and-standards/standards/ecma-335/), section II.25.3.3.
```
Comment (gewarren) [inline xml/System.Reflection.PortableExecutable/PEHeader.xml:356]:
```suggestion
The CLI header contains information and data directories used by the .NET runtime. The CLI header format is defined by the [ECMA-335 specification](https://www.ecma-international.org/publications-and-standards/standards/ecma-335/), section II.25.3.3.
```

File: xml/System.Reflection.PortableExecutable/PEHeader.xml [hunk 2]
```diff
@@ -150,8 +162,8 @@ For program images, this is the starting address. For device drivers, this is th
         <ReturnType>System.Reflection.PortableExecutable.DirectoryEntry</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
-        <value>To be added.</value>
+        <summary>Gets the Base relocations table entry.</summary>
```
Comment (gewarren) [inline xml/System.Reflection.PortableExecutable/PEHeader.xml:165]: I'm unsure about the capitalization here and throughout this file. I see that the field is named "Base Relocations Table" at https://learn.microsoft.com/en-us/windows/win32/debug/pe-format#optional-header-data-directories-image-only. So either all three words should be capitalized, or none of them.

File: xml/System.Reflection.PortableExecutable/PEHeaders.xml [hunk 3]
```diff
@@ -508,7 +508,14 @@
       <Docs>
         <summary>Gets the image's PE header.</summary>
         <value>The image's PE header, or <see langword="null" /> if the image is COFF only.</value>
-        <remarks>To be added.</remarks>
+        <remarks>
+          <format type="text/markdown"><![CDATA[  
+
+## Remarks
+For more information and example, see the <xref:System.Reflection.PortableExecutable.PEHeader> class documentation.
```
Comment (gewarren) [inline xml/System.Reflection.PortableExecutable/PEHeaders.xml:515]:
```suggestion

For more information and an example, see the <xref:System.Reflection.PortableExecutable.PEHeader> class documentation.
```

File: snippets/csharp/System.Reflection.PortableExecutable/PEHeader/Overview/PEHeaderSnippets.csproj [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net6.0</TargetFramework>
```
Comment (gewarren) [inline snippets/csharp/System.Reflection.PortableExecutable/PEHeader/Overview/PEHeaderSnippets.csproj:5]:
```suggestion
    <TargetFramework>net7.0</TargetFramework>
```

File: xml/System.Reflection.PortableExecutable/PEHeader.xml (hunk 2 above)
Comment (MSDN-WhiteKnight) [reply xml/System.Reflection.PortableExecutable/PEHeader.xml]: Changed to capitalize all words in entry names, as preexisting Certificates Table used this style.

**Review by gewarren — APPROVED** [review]
Thanks @MSDN-WhiteKnight

**Pull request merged**